build/index.js

```diff
@@ -35,124 +42,109 @@ const bootstrap = () => {
     if (!fs.existsSync(elmJsonFile)) {
       generateElmJson();
     }
-    return Promise.resolve();
   } catch (e) {
-    return Promise.reject(`Could not complete bootstrap: ${e}`);
+    return new Error(`Could not complete bootstrap: ${e}`);
   }
 };
 
-const fetchIconsArchive = () => {
-  try {
-    const archiveName = "zondicons.zip";
-    const srcArchivePath = `${__dirname}/${archiveName}`;
-    const dstArchivePath = `${tmpPath}/${archiveName}`;
+const wait = (duration) => (...args) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(...args);
+    }, duration);
+  });
+};
 
-    fs.copyFileSync(srcArchivePath, dstArchivePath);
+const fetchIcons = async () => {
+  const stream = require("stream");
+  const { promisify } = require("util");
+  const fetch = require("node-fetch");
+  const unzipper = require("unzipper");
+  const iconsUrl = "http://localhost:8080/zondicons.zip";
+  const asyncStreamPipeline = promisify(stream.pipeline);
 
-    return Promise.resolve(dstArchivePath);
-  } catch (e) {
-    return Promise.reject(`Could not fetch icons archive: ${e}`);
-  }
+  const res = await fetch(iconsUrl);
+
+  return asyncStreamPipeline(res.body, unzipper.Extract({ path: tmpPath }))
+    .then(wait(250))
```

Comment:
:speech_balloon: 
I am using a delay because it seems the unzipper library is reporting it's done extracting earlier than it actually is because if I remove the `wait` only about 75 files are there at the time when the convertion starts.
I went for this hacky way because I want to be able to publish the package and eventually deal with the generation at a later stage (I also have an idea how to improve the generated code and avoid duplication there).

Reply:
No hurry Dimi! You still have some time to fix it ;)

Reply:
I'd also fix it first. 250ms seems kind of arbitrary thus you're running the risk that the stream will still not be done then.

Reply:
Yup, 250 is kind of a magic number here. My point was I wanted to be able to generate the Elm code so I can publish the package and fix the build properly later on since it’s triggered  manually and is not that crucial in this sense.